src/go/rdctl/pkg/plist/plist.go

```diff
@@ -132,13 +139,23 @@ func xmlEscapeText(s string) (string, error) {
 }
 
 // JsonToPlist converts the json settings to a reg file
+// It needs two JSON representations of the incoming data: one based on the schema,
```

Comment:
But it only takes one JSON representation as input. Something is confused here…

If you're trying to explain `settingsJSON` vs `actualSettingsJSON`, that's an implementation detail that should be a comment in the function body rather than the function documentation.

Reply:
I guess it is an implementation detail.

This follows from the decision I made in the rdctl-set code-generator to not declare all the sub-structures in `options.ServerSettingsForJSON` as pointers to structures, and instead relied on having the `serverSettings` type around to actually store the data, and then representing the specified fields in `options.ServerSettingsForJSON` by pointing to the address of data stored in whichever field in `serverSettings`.

Now this doesn't work for arrays or hashes, but we still aren't supporting them in `rdctl set`, so that's not a problem. From my testing, I see that if I declare an array of strings in `options.ServerSettingsForJSON` as 
`List    *[]string `json:"list,omitempty"` both specified empty arrays and unspecified values are both missing.

So I don't regret that decision.

Reply:
Right, the giant hole in my thinking was that the input could be user supplied (`rdctl create-profile --body=…` instead of `rdctl create-profile --from-settings`), so we _do_ have to deal with missing bits.